pkg/kubectl/serviceaccount.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+// ServiceAccountGeneratorV1 supports stable generation of a service account
+type ServiceAccountGeneratorV1 struct {
+	// Name of service account
+	Name string
+}
+
+// Ensure it supports the generator pattern that uses parameter injection
+var _ Generator = &ServiceAccountGeneratorV1{}
```

Comment:
The `StructuredGenerator` interface that you have implemented is enough. Remove anything related to the `Generator` interface and you are set. @deads2k since you rebased the first structured generators in Origin, did you have any problem with them?